### PR TITLE
Support Foreign Key options for MySQL

### LINKED ISSUE #2449: MySQL ON DELETE CASCADE is not supported on FOREIGN KEYS
SQLFluff report `L014 | Unquoted identifiers must be consistently lower case.` on the last two lines of this:

```
CREATE TABLE source_tag_assoc (
    source_id INT UNSIGNED NOT NULL,
    tag_id INT UNSIGNED NOT NULL,
    PRIMARY KEY (source_id, tag_id),
    FOREIGN KEY (source_id) REFERENCES source (id) ON DELETE CASCADE,
    FOREIGN KEY (tag_id) REFERENCES source_tag (id) ON DELETE CASCADE
);
```

but I don't see any problem about cases here.

## Expected Behaviour

No warning on this.

## Observed Behaviour

L014 warning.

## Steps to Reproduce

Run `sqlfluff lint --dialect mysql` on the given example.

## Dialect

Mysql

## Version

```
sqlfluff, version 0.9.2
Python 3.9.9
```

## Configuration
```
Include your SQLFluff configuration here
```

macOS 12.1

### PR DESCRIPTION
**Pull request by tunetheweb**
### Brief summary of the change made

Fixes #2449 

### Are there any other side effects of this change that we should be aware of?
Nope

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_mysql.py [hunk 1]
```diff
@@ -310,7 +310,7 @@ class TableConstraintSegment(BaseSegment):
         Sequence(  # [ CONSTRAINT <Constraint name> ]
             "CONSTRAINT", Ref("ObjectReferenceSegment"), optional=True
         ),
-        OneOf(
+        Delimited(
```
Comment (barrywhart) [inline src/sqlfluff/dialects/dialect_mysql.py:313]: Why was this change needed? (I.e. what test cases / syntax?) From the original issue description, I expected the changes below (line 338...), but not this one.

Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_mysql.py:313]: Pulled in the examples for this link: https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html

It shows comma delimited values are allowed for PRIMARY KEY/INDEX/FOREIGN KEY.

For example (this has been added as a test case in this PR btw):

```sql
CREATE TABLE product_order (
    no INT NOT NULL AUTO_INCREMENT,
    product_category INT NOT NULL,
    product_id INT NOT NULL,
    customer_id INT NOT NULL,

    PRIMARY KEY(no),
    INDEX (product_category, product_id),
    INDEX (customer_id),

    FOREIGN KEY (product_category, product_id)
      REFERENCES product(category, id)
      ON UPDATE CASCADE ON DELETE RESTRICT,

    FOREIGN KEY (customer_id)
      REFERENCES customer(id)
)
```

Comment (barrywhart) [reply src/sqlfluff/dialects/dialect_mysql.py:313]: Got it -- thanks!

**Review by barrywhart — APPROVED** [review]
